File: lab01/builder/src/main/java/com/example/builder/Character.java
package com.example.builder;

import com.example.builder.enums.CharacterType;
import com.example.builder.enums.WeaponType;

import lombok.Getter;

@Getter
public class Character {
    private CharacterType type;
    private String name;
    private String health;
    private String mana;
    private WeaponType weapon;

    public Character(CharacterType type, String name, String health, String mana, WeaponType weapon) {
        this.type = type;
        this.name = name;
        this.health = health;
        this.mana = mana;
        this.weapon = weapon;
    }
}

File: lab01/builder1/Models/Character.cs
using Builder1.Enums;

namespace Builder1.Models;

public class Character
{
    public CharacterType Type { get; }
    public string Name { get; }
    public int? Health { get; }
    public int? Mana { get; }
    public WeaponType? Weapon { get; }

    public Character(CharacterType type, string name, int? health, int? mana, WeaponType? weapon)
    {
        Type = type;
        Name = name;
        Health = health;
        Mana = mana;
        Weapon = weapon;
    }
}
